Integrations/integration-SumoLogic.yml

```diff
@@ -0,0 +1,266 @@
+                    results.records = rec.obj.records.map(function(m) {return m.map;});
+                }
+                return results;
+            }
+            throw 'Timeout while waiting for job ' + res.obj.id;
+        } finally {
+            try {
+                doReq('DELETE', 'search/jobs/' + res.obj.id, null, res.cookies);
+            } catch (ex) {
+                logInfo('SumoLogic error deleting job - ' + ex);
+            }
+        }
+    };
+
+    var a2i = function(v, d) {
+        return v ? parseInt(v) : d;
+    };
+
+    var defaultLimit = 100, defaultSleep = 3, defaultTimeout = 180, defaultSearchTimeout = 10;
+
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            doReq('GET', 'collectors', {limit: '1'}, null);
+            return 'ok';
+        case 'fetch-incidents':
+            if (!params.fetchQuery) {
+                throw 'No fetch query defined, not doing SumoLogic fetch';
+            }
+            var now = (new Date()).getTime();
+            var lastRun = getLastRun();
+            if (!lastRun || !lastRun.time) {
+                // First time, retrieve events from the last 10 min
+                lastRun = {time: now - 10 * 60 * 1000};
+            }
+            var s = search(params.fetchQuery, lastRun.time, now, a2i(params.limit, defaultLimit), 0, 'UTC', a2i(params.maxTimeout, 180),
+                a2i(params.sleepBetweenChecks, defaultSleep));
+            var incidents = [];
+            if (s.messages) {
+                for (var i=0; i<s.messages.length; i++) {
+                    var incident = {name: 'Incident from SumoLogic ' + s.messages[i]._messageid,
+                        details: s.messages[i]._raw, labels: [], rawJSON: JSON.stringify(s.messages[i])};
+                    var props = Object.getOwnPropertyNames(s.messages[i]);
+                    for (var j=0; j<props.length; j++) {
+                        if (props[j] !== '_raw') {
+                            incident.labels.push({type: props[j], value: s.messages[i][props[j]]});
+                        }
+                    }
+                    incidents.push(incident);
+                }
+            }
+            setLastRun({time: now});
+            return JSON.stringify(incidents);
+        case 'search':
+            var s = search(args.query, args.from, args.to, a2i(args.limit, defaultLimit), a2i(args.offset, 0), args.timezone,
+                a2i(args.maxTimeToWaitForResults, defaultSearchTimeout) * 60, a2i(params.sleepBetweenChecks, defaultSleep));
+            var md = '';
+            var ec = {};
+            if (s.messages && s.messages.length > 0) {
+                md = tableToMarkdown('Messages', s.messages) + '\n';
+                ec.Search = {Messages: s.messages.length > defaultLimit ? s.messages.slice(0, defaultLimit) : s.messages};
```

Comment:
If i will run the same command twice in war room what will happen to the context ? will it override or duplicate ?